operators/src/plot/temporal_vector_line_plot.rs

```diff
@@ -0,0 +1,360 @@
+        let input_processor = self.vector_sources[0].query_processor()?;
+
+        let processor = call_on_generic_vector_processor!(input_processor, features => {
+            FeatureAttributeValuesOverTimeQueryProcessor { params: self.state.clone(), features }.boxed()
+        });
+
+        Ok(TypedPlotQueryProcessor::JsonVega(processor))
+    }
+}
+
+/// A query processor that calculates the `TemporalVectorLinePlot` on its input.
+pub struct FeatureAttributeValuesOverTimeQueryProcessor<G>
+where
+    G: Geometry + ArrowTyped + Sync + Send + 'static,
+{
+    params: FeatureAttributeValuesOverTimeParams,
+    features: Box<dyn VectorQueryProcessor<VectorType = FeatureCollection<G>>>,
+}
+
+#[async_trait]
+impl<G> PlotQueryProcessor for FeatureAttributeValuesOverTimeQueryProcessor<G>
+where
+    G: Geometry + ArrowTyped + Sync + Send + 'static,
+{
+    type OutputFormat = PlotData;
+
+    fn plot_type(&self) -> &'static str {
+        FEATURE_ATTRIBUTE_OVER_TIME_NAME
+    }
+
+    async fn plot_query<'a>(
+        &'a self,
+        query: QueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> Result<Self::OutputFormat> {
+        let values = FeatureAttributeValues::<MAX_FEATURES>::default();
+
+        let values = self
+            .features
+            .vector_query(query, ctx)?
+            .fold(Ok(values), |acc, features| async {
+                match (acc, features) {
+                    (Ok(mut acc), Ok(features)) => {
+                        let ids = features.data(&self.params.id_column)?;
+                        let values = features.data(&self.params.value_column)?;
+
+                        for i in 0..features.len() {
+                            let id: String = match ids.get_unchecked(i) {
+                                FeatureDataValue::Int(v) => v.to_string(),
+                                FeatureDataValue::NullableInt(v) => v
+                                    .map(|v| v.to_string())
+                                    .ok_or(Error::FeatureDataValueMustNotBeNull)?,
+                                FeatureDataValue::Float(v) => v.to_string(),
+                                FeatureDataValue::NullableFloat(v) => v
+                                    .map(|v| v.to_string())
+                                    .ok_or(Error::FeatureDataValueMustNotBeNull)?,
+                                FeatureDataValue::Text(v) => v.clone(),
+                                FeatureDataValue::NullableText(v) => {
+                                    v.ok_or(Error::FeatureDataValueMustNotBeNull)?
+                                }
+                                FeatureDataValue::Category(v) => v.to_string(),
+                                FeatureDataValue::NullableCategory(v) => {
+                                    v.map(|v| v.to_string())
+                                        .ok_or(Error::FeatureDataValueMustNotBeNull)?
+                                }
```

Comment:
What about a method `as_string()`?

I guess we will need it elsewhere and it improves readability.

Reply:
tbh I much rather would like a way of iterating over a column as a certain type. So turning a FeatureDataRef into an Iterator that automatically converts int to float or int to string etc. I created a card https://trello.com/c/MSNIEFxd

Reply:
Could you extract a function then to improve the readability of the function and then put a link from the trello card to this code block such that we know what to replace once we created the functionality?

Reply:
92801e562d20b900275516583b8b8ab4b228334b